source/pod.go

```diff
@@ -154,7 +154,28 @@ func (ps *podSource) Endpoints(ctx context.Context) ([]*endpoint.Endpoint, error
 		if ps.podSourceDomain != "" {
 			domain := pod.Name + "." + ps.podSourceDomain
 			if len(targets) == 0 {
-				addToEndpointMap(endpointMap, domain, suitableType(pod.Status.PodIP), pod.Status.PodIP)
+				ept := EndpointsTypePodIP
+				if tmp, ok := pod.Annotations[endpointsTypeAnnotationKey]; ok {
+					ept = tmp
+				}
+
+				switch ept {
+				case EndpointsTypeNodeExternalIP:
+					node, _ := ps.nodeInformer.Lister().Get(pod.Spec.NodeName)
+					for _, address := range node.Status.Addresses {
```

Comment:
Does not seems right to ignore error.

Reply:
this was copied from slightly above https://github.com/kubernetes-sigs/external-dns/blob/97626894c2d0129ac61b65e09ef8c02ee66b7953/source/pod.go#L115

Reply:
Yeah, most likely low risk. I leave it here https://github.com/kubernetes/apiserver/blob/9b509bf53bc25f12de99091491d95876f89f67a6/pkg/admission/plugin/policy/internal/generic/lister.go#L44

If function will throw an error (not sure what the conditions are, probably likelhood is low) the external dns will crash due to null pointer.